serverless/src/shared/cleanupJsonObject.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Cleans up a JSON object by removing empty strings, null values, and empty objects or arrays.
+ *
+ * @param {Object|Array} obj - The object or array to clean up.
+ * @returns {Object|Array} A new object or array with empty values removed.
+ *
+ * @example
+ * // Cleaning up a simple object
+ * const simpleObj = {
+ *   name: "John",
+ *   age: 30,
+ *   email: "",
+ *   phone: null
+ * };
+ * console.log(cleanupJsonObject(simpleObj));
+ * // Output: { name: "John", age: 30 }
+ *
+ * @example
+ * // Cleaning up a nested object
+ * const nestedObj = {
+ *   user: {
+ *     id: 1,
+ *     details: {
+ *       firstName: "Alice",
+ *       lastName: "Smith",
+ *       middleName: ""
+ *     },
+ *     preferences: {}
+ *   },
+ *   settings: null
+ * };
+ * console.log(cleanupJsonObject(nestedObj));
+ * // Output: { user: { id: 1, details: { firstName: "Alice", lastName: "Smith" } } }
+ *
+ * @example
+ * // Cleaning up an object with arrays
+ * const objWithArrays = {
+ *   prefLabel: "KB",
+ *   root: false,
+ *   numberOfCollections: null,
+ *   scheme: "DistributionSizeUnit",
+ *   version: "20.6",
+ *   uuid: "dfa902e9-a19e-4f85-9cb3-983ddff45011",
+ *   fullPath: "KB",
+ *   longName: "",
+ *   narrower: [],
+ *   broader: {
+ *     prefLabel: "Distribution Size Unit",
+ *     uuid: "e0e6f883-9dee-4908-bb76-20adae968df1"
+ *   }
+ * };
+ * console.log(cleanupJsonObject(objWithArrays));
+ * // Output:
+ * // {
+ * //   prefLabel: "KB",
+ * //   root: false,
+ * //   scheme: "DistributionSizeUnit",
+ * //   version: "20.6",
+ * //   uuid: "dfa902e9-a19e-4f85-9cb3-983ddff45011",
+ * //   fullPath: "KB",
+ * //   broader: {
+ * //     prefLabel: "Distribution Size Unit",
+ * //     uuid: "e0e6f883-9dee-4908-bb76-20adae968df1"
+ * //   }
+ * // }
+ */
+export const cleanupJsonObject = (obj) => {
```

Comment:
You may want to just considering using lodash compactDeep, I think it does the same thing. https://github.com/nasa/mmt/blob/97d5f3cfdd6ec62e9542ac4c6cec40be8dac431d/static/src/js/utils/removeEmpty.js#L9

Reply:
Done